sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinExec.scala

```diff
@@ -206,10 +213,19 @@ case class StreamingSymmetricHashJoinExec(
     val updateStartTimeNs = System.nanoTime
     val joinedRow = new JoinedRow
 
+    // Filter the joined rows based on the given condition.
+    val leftPreJoinFilter =
+      newPredicate(condition.leftSideOnly.getOrElse(Literal(true)), output).eval _
```

Comment:
I am not sure this should be a predicate on the output. We are applying this on the input row, so predicate must be calculated on that rows attributes, which in this case is, left.output.

I wonder why this isnt failing any test.

Reply:
Also, you can dedup the code a little bit, but pushing the code generation into the Joiner. This would also be consistent with other inputs to the joiner (as expressions, not as function).

Reply:
Moving it in would require either also passing in the left ++ right input attributes, or passing preJoin and postJoin filters differently. I'm not sure which option is cleaner, so I can make the change you think is best.

Reply:
preJoin: Joiner.inputAttributes should be sufficient.
postJoin: Just use the overall output.